src/evaluator.ts

```diff
@@ -154,27 +234,61 @@ export class Evaluator {
       }
 
       // evaluation succeded, find variation in flag
-      return this.findVariation(fc, identifier);
+      return identifier;
     }
     // there is no rules return undefined
     return undefined;
   }
 
-  private evaluateFlag(fc: FeatureConfig, target: Target): Variation | undefined {
-    if (fc.state === FeatureState.Off) {
-      return this.findVariation(fc, fc.offVariation);
-    }
-    let variation = this.evaluateRules(fc, target);
-    if (!variation) {
-      // if variation is undefined then use flag default serve
-      if (fc.defaultServe.distribution) {
-        const identifier = this.getKeyName(fc.defaultServe.distribution, target);
-        variation = this.findVariation(fc, identifier);
-      } else {
-        variation = this.findVariation(fc, fc.defaultServe.variation);
+  private evaluateVariationMap(
+    variationToTargetMap: VariationMap[],
+    target: Target,
+  ): string | undefined {
+    if (!variationToTargetMap) return undefined;
+
+    for (const variationMap of variationToTargetMap) {
+      // find target
+      const targetMap = variationMap.targets?.find(
+        (elem) => elem.identifier === target.identifier,
+      );
+      if (targetMap) return variationMap.variation;
+
+      // find target in segment
+      const segmentIdentifiers = variationMap.targetSegments;
+      if (
+        segmentIdentifiers &&
+        this.isTargetIncludedOrExcludedInSegment(segmentIdentifiers, target)
+      ) {
+        return variationMap.variation;
       }
     }
-    return variation
+
+    return undefined;
```

Comment:
Again, no real need to specifically return `undefined`. Not returning will do the same thing. No need to change

Reply:
compiler complains